### DISCUSSION TITLE
Vegetables as porn?

### DISCUSSION THREAD

**Discussion by Saymon1998** [infinitered/nsfwjs#702] [Model Training]
Hi everyone, lately I came into an issue where mobilenetV1 model (2,6 MB) gave me most strange output when I send an image with onions to classify. I did check it also on other vegetables, like apples and beetroot, and it looks like that only apples aren't classified as porn/hentai :) 
Onions gave me results:
0: {className: "Porn", probability: 0.7866919040679932}
1: {className: "Neutral", probability: 0.14794990420341492}
2: {className: "Sexy", probability: 0.04114266857504845}
3: {className: "Hentai", probability: 0.022164924070239067}
4: {className: "Drawing", probability: 0.0020506654400378466}

Apples:
0: {className: "Neutral", probability: 0.9997453093528748}
1: {className: "Drawing", probability: 0.00016232636698987335}
2: {className: "Porn", probability: 0.00005498077371157706}
3: {className: "Sexy", probability: 0.00002530598248995375}
4: {className: "Hentai", probability: 0.000012140934813942295}

Beetroot:
0: {className: "Hentai", probability: 0.41282495856285095}
1: {className: "Drawing", probability: 0.302909255027771}
2: {className: "Neutral", probability: 0.20993521809577942}
3: {className: "Porn", probability: 0.05640846863389015}
4: {className: "Sexy", probability: 0.01792212948203087}

I wonder, why? Does mobilenet model react like that because of onions colour? Maybe beetroots roots somehow resemble tentacles in some way, at least, for this model?

Used images:
![apples](https://user-images.githubusercontent.com/61495756/166651423-17512937-cd1b-4591-9cc1-67c31748c8a6.jpeg)
![onions](https://user-images.githubusercontent.com/61495756/166651439-f4bca9bc-fc21-4b33-8d97-31a194e1056f.jpg)
![beetroots](https://user-images.githubusercontent.com/61495756/166651448-2baa086e-194d-4115-a586-6023e1e34c0f.jpg)

Cheers

**Comment by 952xuan** [comment]
White and reflective Mantous(Chinese steamed buns) can be more pornographic.  Maybe we should do human target detection before using this model.

**Comment by GantMan** [comment]
thank you for this - will add to training
